packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -376,7 +383,11 @@ export function withExtendedResolver(
 
   const metroConfigWithCustomResolver = withMetroResolvers(config, [
     // Mock out production react imports in development.
-    (context: ResolutionContext, moduleName: string, platform: string | null) => {
+    function requestDevMockProdReact(
```

Comment:
I gave the functions here names, since it was useful while debugging when I was adding extra logs and observing whether a prior one threw. It's unrelated to this PR, but could be nice to keep. Revertable via this commit: https://github.com/expo/expo/pull/34286/commits/8e722f26f50e1a89555154fc175be4624544789e